entries/block.go

```diff
@@ -189,14 +156,27 @@ func bulkInsertBlockEntry(entries []*lib.StateChangeEntry, db bun.IDB, operation
 	blockQuery := db.NewInsert().Model(&pgBlockEntrySlice)
 
 	if operationType == lib.DbOperationTypeUpsert {
-		blockQuery = blockQuery.On("CONFLICT (block_hash) DO UPDATE")
+		blockQuery = blockQuery.On(`CONFLICT (height) DO UPDATE
+			SET prev_block_hash = EXCLUDED.prev_block_hash,
+				block_hash = EXCLUDED.block_hash,
+				txn_merkle_root = EXCLUDED.txn_merkle_root,
+				timestamp = EXCLUDED.timestamp,
+				nonce = EXCLUDED.nonce,
+				extra_nonce = EXCLUDED.extra_nonce,
+				block_version = EXCLUDED.block_version,
+				proposer_voting_public_key = EXCLUDED.proposer_voting_public_key,
+				proposer_random_seed_signature = EXCLUDED.proposer_random_seed_signature,
+				proposed_in_view = EXCLUDED.proposed_in_view,
+				proposer_vote_partial_signature = EXCLUDED.proposer_vote_partial_signature
+				WHERE handle_block_conflict(pg_block_entry.block_hash) IS NOT NULL
```

Comment:
If the existing block hash matches, we don't want do to do the deletions, right?